comp/autoscaling/datadogclient/datadogclientimpl/client_mock.go

```diff
@@ -0,0 +1,66 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build test
+
+package datadogclientimpl
+
+import (
+	"sync"
+
+	"github.com/DataDog/datadog-agent/comp/autoscaling/datadogclient"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+	"gopkg.in/zorkian/go-datadog-api.v2"
+)
+
+// MockModule defines the fx options for the mock component.
+func MockModule() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newMock),
+	)
+}
+
+type mockDatadogClient struct {
+	mux               sync.RWMutex
+	queryMetricsFunc  func(from, to int64, query string) ([]datadog.Series, error)
+	getRateLimitsFunc func() map[string]datadog.RateLimit
+}
+
```

Comment:
nit: maybe you can add a local var to verify that the mock implement the interface, like you did for the implement

```
var _ datadogclient.Component = (*mockDatadogClient)(nil)
```